### PR TITLE
Targets unshared, again

### PR DESCRIPTION
**Pull request by toddburnside**
Initial changes for the unshared targets. Much remains to be done, for example:

- Fix and re-add the target summary table.
- For new targets, create the id locally so we don't have to wait for the round trip.
- Show the target list on the targets tab, and allow dragging to asterisms - and potentially editing the targets directly.

I will do some more renaming of classes/values in later PRs. There seemed to be plenty of changes for one PR...

There are still some issues with the updating the tree locally after adding/removing targets from asterisms. I will continue looking into that in the morning. Everything gets resolved after the server roundtrip, so we can merge before that is fixed if we'd like.

### CODE REVIEW

File: common-graphql/src/main/scala/explore/common/TargetListGroupQueriesGQL.scala [hunk 1]
```diff
@@ -20,52 +21,58 @@ object TargetListGroupQueriesGQL {
   trait TargetListGroupObsQuery extends GraphQLOperation[ObservationDB] {
```
Comment (rpiaggio) [inline common-graphql/src/main/scala/explore/common/TargetListGroupQueriesGQL.scala:21]: Should we rename this to `AsterismGroupObsQuery` ?

File: common-graphql/src/main/scala/explore/common/TargetListGroupQueriesGQL.scala [hunk 2]
```diff
@@ -113,9 +124,20 @@ object TargetListGroupQueriesGQL {
   @GraphQL
   trait ReplaceScienceTargetListMutation extends GraphQLOperation[ObservationDB] {
```
Comment (rpiaggio) [inline common-graphql/src/main/scala/explore/common/TargetListGroupQueriesGQL.scala:125]: `UpdateTargetEnvironmentMutation`?

File: common/src/main/scala/explore/common/TargetListGroupQueries.scala [hunk 3]
```diff
@@ -14,64 +14,52 @@ import crystal.react.reuse._
...
 import lucuma.ui.reusability._
 import monocle.Focus
 import monocle.Getter
 
 import scala.collection.immutable.SortedMap
+import scala.collection.immutable.SortedSet
 
 import TargetListGroupQueriesGQL._
 
 object TargetListGroupQueries {
```
Comment (rpiaggio) [inline common/src/main/scala/explore/common/TargetListGroupQueries.scala:37]: More pontential `TargetList` -> `Asterism` renames.

File: common/src/main/scala/explore/common/TargetQueries.scala [hunk 4]
```diff
@@ -33,61 +37,72 @@ import TargetQueriesGQL._
...
         )
 
     def apply[A](
       modelLens: Lens[SiderealTarget, A],
-      remoteSet: A => EditSiderealInput => EditSiderealInput
+      remoteSet: A => EditTargetInput => EditTargetInput
     ): View[A] = apply(modelLens.get, modelLens.modify, remoteSet)
   }
 
+  def toTargetEndo(setSidereal: Endo[EditSiderealInput]): Endo[EditTargetInput] = eti =>
```
Comment (rpiaggio) [inline common/src/main/scala/explore/common/TargetQueries.scala:67]: Nice 👍

File: common/src/main/scala/explore/common/TargetListGroupQueries.scala (hunk 3 above)
Comment (toddburnside) [reply common/src/main/scala/explore/common/TargetListGroupQueries.scala:37]: Yeah. As I noted in the description, I'm going to do more renaming. But, it seemed like the diff was going to be hard enough to follow - file renaming doesn't get tracked well in git. 
 
I will make another commit (in this PR) with a bunch of renaming.

File: explore/src/main/scala/explore/targeteditor/TargetEnvEditor.scala [hunk 5]
```diff
@@ -27,17 +27,18 @@ import lucuma.core.model.SiderealTarget
...
 import lucuma.core.model.User
 import lucuma.ui.reusability._
-import monocle.function.Index._
+import monocle.Optional
 import react.common.ReactFnProps
 import react.semanticui.elements.button._
 import react.semanticui.shorthand._
 import react.semanticui.sizes._
 
 final case class TargetEnvEditor(
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targeteditor/TargetEnvEditor.scala:36]: Maybe we should rename `TargetEnv` to something else? Or do we still handle that concept?

File: explore/src/main/scala/explore/targets/TargetSelectionPopup.scala [hunk 6]
```diff
@@ -44,7 +44,7 @@ import scala.concurrent.duration._
 
 final case class TargetSelectionPopup(
   trigger:          Reuse[Button],
-  onSelected:       Target ==> Callback
+  onSelected:       (Option[Target.Id], Target) ==> Callback
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targets/TargetSelectionPopup.scala:47]: Nice

File: model/shared/src/main/scala/explore/model/util/NonEmptySetWrapper.scala [hunk 7]
```diff
@@ -74,6 +77,9 @@ final class NonEmptySetWrapper[IdSet, Id](self: IdSet, iso: Iso[IdSet, NonEmptyS
   @inline
   def intersects(other: IdSet): Boolean = intersect(other).nonEmpty
 
+  @inline
+  def firstAndOnly: Option[Id] = if (selfSet.length === 1) selfSet.head.some else none
```
Comment (rpiaggio) [inline model/shared/src/main/scala/explore/model/util/NonEmptySetWrapper.scala:81]: `single` ?

File: model-testkit/shared/src/main/scala/explore/model/arb/ArbObsSummary.scala [hunk 8]
```diff
@@ -32,14 +31,13 @@ trait ArbObsSummary {
...
       duration     <- arbitrary[Duration]
-      targetEnvId  <- arbitrary[TargetEnvironment.Id]
       targets      <- arbitrary[Set[Target.Id]]
     } yield ObsSummaryWithConstraints(id,
                                       constraints,
                                       status,
                                       activeStatus,
                                       duration,
-                                      targetEnvId,
+                                      // targetEnvId,
```
Comment (rpiaggio) [inline model-testkit/shared/src/main/scala/explore/model/arb/ArbObsSummary.scala:40]: Commented code

File: explore/src/main/scala/explore/targets/TargetSelectionPopup.scala (hunk 6 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/targets/TargetSelectionPopup.scala:47]: Should we have case class for this? Seems to be used in a few places.

**Review by rpiaggio — APPROVED** [review]
LGTM

Looks great, thank you for doing this! 

I like the feeling of having a lighter mental load.

File: common/src/main/scala/explore/common/TargetListGroupQueries.scala (hunk 3 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/common/TargetListGroupQueries.scala:37]: :+1: I missed that line in the description

File: explore/src/main/scala/explore/targeteditor/TargetEnvEditor.scala (hunk 5 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/targeteditor/TargetEnvEditor.scala:36]: I think it should probably be the AsterismEditor.

File: model/shared/src/main/scala/explore/model/util/NonEmptySetWrapper.scala (hunk 7 above)
Comment (toddburnside) [reply model/shared/src/main/scala/explore/model/util/NonEmptySetWrapper.scala:81]: Sounds good. I couldn't think of a good name at the moment I created it....

File: explore/src/main/scala/explore/targets/TargetSelectionPopup.scala (hunk 6 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/targets/TargetSelectionPopup.scala:47]: Yeah, might be more clear. `TargetWithId` is actually used more places and is just an alias for a similar tuple. I've been wondering if that, along with `SiderealTargetWithId` and `NonsiderealTargetWithId`, should be bumped up to a case class.

**Pull request merged**